app/models/deploy_service.rb

```diff
@@ -47,6 +48,17 @@ def confirm_deploy(deploy)
 
   private
 
+  def update_average_deploy_time(deploy)
+    stage = deploy.stage
+
+    old_average = stage.average_deploy_time || 0.00
+    number_of_deploys = stage.deploys.size
+
+    new_average = (old_average + ((deploy.duration - old_average) / number_of_deploys))
+
+    stage.update_column(:average_deploy_time, new_average)
+  end
```

Comment:
kinda sketchy math ... can you explain why this is needed or switch to something simple like average from last 10 ?

Reply:
It gives a more accurate picture of the overall average deploy time of that particular stage. If there were 5 failed deploys or more for a particular stage because of some bad config or something, then the average would be greatly skewed. That might be pretty confusing to some people when their stage ETA changes so drastically due to a couple of failed or hung deploys, even though they might have hundreds of past deploys with normal run times.